src/nplinker/metabolomics/gnps/gnps_annotation_loader.py

```diff
@@ -1,42 +1,134 @@
+            'json_url': 'https://metabolomics-usi.gnps2.org/json/?usi1=mzspec:GNPS:GNPS-LIBRARY:accession:CCMSLIB00000086167',
+            'svg_url': 'https://metabolomics-usi.gnps2.org/svg/?usi1=mzspec:GNPS:GNPS-LIBRARY:accession:CCMSLIB00000086167',
+            'spectrum_url': 'https://metabolomics-usi.gnps2.org/spectrum/?usi1=mzspec:GNPS:GNPS-LIBRARY:accession:CCMSLIB00000086167'}
         """
         self._file = Path(file)
-        self._annotations : dict[str, dict] = {}
+        self._annotations: dict[str, dict] = {}
 
-        with open(self._file, mode='rt', encoding='UTF-8') as f:
-            header = f.readline().split('\t')
-            dict_reader = csv.DictReader(f, header, delimiter='\t')
-            for row in dict_reader:
-                scan_id = row.pop('#Scan#')
-                self._annotations[scan_id] = row
+        self._validate()
+        self._load()
 
-                # also insert useful URLs
-                for t in ['png', 'json', 'svg', 'spectrum']:
-                    self._annotations[scan_id][f'{t}_url'] = GNPS_URL_FORMAT.format(t, row['SpectrumID'])
+    @property
+    def annotations(self) -> dict[str, dict]:
+        """Get annotations.
 
+        Returns:
+            dict[str, dict]: Keys are spectrum ids ("#Scan#" in annotation file)
+                and values are the annotations dict for each spectrum.
+        """
+        return self._annotations
 
+    def _validate(self) -> None:
+        """Validate the annotation file.
 
-    def get_annotations(self) -> dict[str, dict]:
-        """Get annotations.
+        Raises:
+            ValueError: Raises ValueError if the file is not valid.
+        """
+        # validate file format
+        if not is_file_format(self._file, 'tsv'):
+            raise ValueError(f"Invalid GNPS annotation file '{self._file}'. "
+                             f"Expected a .tsv file.")
 
-        Returns:
-            dict[str, dict]: Spectra indices are keys and values are the annotations for this spectrum.
+        # validate required columns against the header
+        required_columns = [
+            '#Scan#', 'Compound_Name', 'Organism', 'MQScore', 'SpectrumID'
+        ]
+        with open(self._file, mode='rt') as f:
+            header = f.readline()
+            for k in required_columns:
+                if k not in header:
+                    raise ValueError(
+                        f"Invalid GNPS annotation file '{self._file}'. "
+                        f"Expected a header line with '{k}' column, "
+                        f"but got '{header}'.")
 
-        Examples:
-            >>> print(loader.annotations()[100])
-            """
-        return self._annotations
+        # validate that "#Scan#" must be unique
+        with open(self._file, mode='rt') as f:
+            reader = csv.DictReader(f, delimiter='\t')
+            ids = []
```

Comment:
What is the purpose of this list?

Reply:
It acts as a temporary container. This validation part is not intuitive, will update it.